src/test/integration/handlers/challenge/FindAllChallengesTest.kt

```diff
@@ -0,0 +1,80 @@
+package test.integration.handlers.challenge
+
+import com.amazonaws.services.lambda.runtime.Context
+import com.beust.klaxon.JsonObject
+import com.beust.klaxon.Klaxon
+import framework.models.idValue
+import io.kotlintest.specs.WordSpec
+import io.mockk.junit5.MockKExtension
+import org.junit.jupiter.api.extension.ExtendWith
+import io.kotlintest.Description
+import io.kotlintest.shouldBe
+import io.kotlintest.TestResult
+import kotlinserverless.framework.models.Handler
+import io.mockk.mockk
+import main.daos.*
+import main.helpers.JsonHelper
+import test.TestHelper
+import org.jetbrains.exposed.sql.transactions.transaction
+
+@ExtendWith(MockKExtension::class)
+class FindAllChallengesTest : WordSpec() {
+    private lateinit var handler: Handler
+    private lateinit var contxt: Context
+    private lateinit var user1: UserAccount
+    private lateinit var user2: UserAccount
+    private lateinit var challenge: Challenge
+    private lateinit var map: Map<String, Any>
+    private lateinit var notFoundMap: Map<String, Any>
+
+    override fun beforeTest(description: Description) {
+        Handler.connectAndBuildTables()
+        handler = Handler()
+        contxt = mockk()
+        transaction {
+            val newUsers = TestHelper.generateUserAccounts(2)
+            user1 = newUsers[0].value
+            user2 = newUsers[1].value
+            challenge = TestHelper.generateFullChallenge(user1, user1).first()
+            map = mutableMapOf(
+                    Pair("path", "/challenge/findAll"),
```

Comment:
i wonder if this needs to have findall...

so can you also try 1 thing before using this route

try with /challenge without passing an ID and also update Controller.defaultRouting
for the HTTP_GET when -> else
swicth from throwing to:
restController.findAll(user, getRawBody(request) as Map<String, String>)

and update ReadabilityController to have a new method like

fun findAll(user: User, filters: Map<String, Any> = emptyMap()): SOAResult<T> {
invalidendpoint()

etc...

and also would require you to update the ChallengeController method to override this...


can you try this out? might be a ton cleaner